src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1076,9 +1076,8 @@ bool Compiler::CheckHWIntrinsicImmRange(NamedIntrinsic intrinsic,
 
         if (immOutOfRange)
         {
-            assert(!mustExpand);
-            // The imm-HWintrinsics that do not accept all imm8 values may throw
-            // ArgumentOutOfRangeException when the imm argument is not in the valid range
+            // We'll use a fallback throwing ArgumentOutOfRangeException
+            *useFallback = true;
```

Comment:
Are we losing or missing context here that the call will definitively throw? Maybe that can help with control flow and tagging things as hot/cold?